src/corppa/poetry_detection/passim/create_passim_corpus.py

```diff
@@ -0,0 +1,177 @@
+"""
+Create a passim-friendly version of the input corpus.
+"""
+
+import argparse
+import pathlib
+import sys
+from typing import Iterator
+
+import orjsonl
+from tqdm import tqdm
+
+
+def transform_record(
+    record: dict,
+    corpus_name: str,
+    id_field: str = "id",
+    preserve_fields: bool = False,
+):
+    """
+    Converts a record (dict) to a passim-friendly form.
+
+    This method returns a new record (dict) with the following fields:
+        * "id": record[id_field]
+        * "corpus": corpus_name
+        * "text": record["text"]
+
+    Optionally, the output can preserve all fields found in the input records by
+    setting preserve_fields to True.
+    """
+    # Validate input record
+    if id_field not in record:
+        raise ValueError(f"Record missing '{id_field}' field")
+
+    out_record = {}
+    # Copy over all fields
+    if preserve_fields:
+        if id_field != "id" and "id" in record:
+            raise ValueError(f"Record has existing 'id' field")
+        out_record |= record
+    # Add required output
+    out_record["id"] = record[id_field]
+    out_record["corpus"] = corpus_name
+    # If text field is missing, treat as blank page
+    out_record["text"] = record.get("text", "")
```

Comment:
I mean, these files were adjudicated? It's not like it takes any space, I find it more troubling if a transformation drops records. I explicitly had to include these cases because the PPA text corpus has pages without "text" fields, but we included them in our annotation rounds.

Reply:
I've made a note so we can revisit this in the future.